### ISSUE TITLE
NullPointerException in unregisterNoisyReceiver

### ISSUE THREAD

**Issue by ened** [ryanheise/audio_session#13]
```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.content.Context.unregisterReceiver(android.content.BroadcastReceiver)' on a null object reference
       at com.ryanheise.audio_session.AndroidAudioManager.unregisterNoisyReceiver(AndroidAudioManager.java:110)
       at com.ryanheise.audio_session.AndroidAudioManager.abandonAudioFocus(AndroidAudioManager.java:85)
       at com.ryanheise.audio_session.AndroidAudioManager.lambda$requestAudioFocus$0(AndroidAudioManager.java:66)
       at com.ryanheise.audio_session.-$$Lambda$AndroidAudioManager$vTsUKVRsAvXdPLcXpQfkDXFTmrU.onAudioFocusChange(-.java:2)
       at android.media.AudioManager$ServiceEventHandlerDelegate$1.handleMessage(AudioManager.java:2886)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:223)
       at android.app.ActivityThread.main(ActivityThread.java:7660)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
```

Happens when `applicationContext` became `null`.

This is in the latest version 0.0.10.

**Comment by ryanheise** [comment]
Hi @ened , thanks for the report. I will look into it.

**Comment by ryanheise** [comment]
Can you try the latest commit? Tracing the logic that could cause this, I would guess your app has at least 2 FlutterEngines/isolates, and the second one was shutdown, followed by a loss of audio focus. The other possibility (which doesn't seem to be the case here) is that the first isolate was shut down. The latest commit should hopefully handle both situations, although since I wasn't actually able to reproduce the error, hopefully you can try it out and let me know how it goes.

**Comment by ened** [comment]
2 isolates, correct - either of which can be stopped at any time and started at any time (Android Activity + Foreground Service).
This is happening in our production App - appreciate if you can cut a new release for this.

**Comment by ryanheise** [comment]
Just to confirm, did the latest commit work for you?

**Comment by ryanheise** [comment]
Note that I would normally want to do more testing before publishing a new release with a substantial change to the logic like this. However, if you are happy with that commit, I would recommend until then that you pin your app to that specific commit as follows:

```yaml
dependencies:
  audio_session:
    git:
      url: https://github.com/ryanheise/audio_session.git
      ref: e40d76dcda985fea4a42c06839f30f37af1c7781
```

(You may need `dependency_overrides` if you are also using other plugins that use audio_session such as [audio_service](https://pub.dev/packages/audio_service) or [just_audio](https://pub.dev/packages/just_audio).)

**Comment by ened** [comment]
@ryanheise looks good - thank you for your help. I've tested this on our app locally and did not observe any crashes.

**Comment by ryanheise** [comment]
Great! I'll probably do the next release this coming weekend. If you encounter any further issues during that time, let me know and I'll try to address them before the release.

**Comment by ened** [comment]
@ryanheise Hi are you still planning a release?

**Comment by ryanheise** [comment]
Sure, I can do that now; I assume you didn't run into any further crashes with this commit?

**Comment by ened** [comment]
Yes looks good so far. We will pass the release candidate to our testers this week - will report if more issues are found.

**Comment by ryanheise** [comment]
Thanks, I also haven't run into any issues on my end so I've published 0.0.11.

I'll close this for now but if you do run into further issues, do let me know and I will reopen.
